### ISSUE TITLE
Replace `only-needed` bool by a `mode` selector with multiple values

### ISSUE THREAD

**Issue by fingolfin** [gap-actions/run-pkg-tests#29]
E.g.
- `mode: default` -- current default behaviour (just load GAP with default set of package then run tests)
- `mode: onlyneeded` -- like `only-needed: true`
- `mode: loadall` -- perform `LoadAllPackages` before running the actual tests
   - however in general this requires all packages to be compiled (or at least as many as possible)

**Comment by fingolfin** [comment]
While in #38 @Joseph-Edwards describes that we can't have a multivalue "selector", I still think it is possible to implement this, as a string input, with the list possible values, no? And ideally any other value would lead to an error.

**Comment by Joseph-Edwards** [comment]
Yeah this is definitely something I think should work. I'll have a crack at that now.
